feat: 페이지 단위로 Study정보 조회할 수 있도록 백엔드 구현 (#91)

Deocksoo opened the pull request:
커밋은 Repository부터 남겼지만 구현은 ATDD로 하였습니다.

Deocksoo committed:
test: StudyRepositoryTest에 page 단위 조회 테스트 추가

Deocksoo committed:
feat: StudyService에 page단위로 StudyInfoDto 찾아오는 기능 추가

Deocksoo committed:
feat: StudyApiController에 page단위 조회 기능 추가

Deocksoo committed:
refactor: Study 관련 dto 패키지 구조 변경

ddu0422 commented on src/main/java/woowahan/anifarm/tecolearning/study/service/StudyService.java:
```diff
@@ -36,6 +41,13 @@ public StudyInfoDto findInfoDtoById(long studyId) {
+        Page<Study> pageOfStudy = studyRepository.findAll(pageable);
+        return pageOfStudy.stream()
+                .map(StudyInfoDto::from)
+                .collect(Collectors.toList());
```
Collections.unmodifiableList()를 사용해 불변으로 만드는건 어떤가요?

ddu0422 commented on src/main/java/woowahan/anifarm/tecolearning/study/controller/StudyApiController.java:
```diff
@@ -29,6 +32,12 @@ public StudyApiController(StudyService studyService) {
+    @GetMapping
+    public ResponseEntity<List<StudyInfoDto>> readOnePage(Pageable pageable) {
+        List<StudyInfoDto> studyInfoDtos = studyService.findPageOfInfoDto(pageable);
+        return ResponseEntity.ok(studyInfoDtos);
```
다른 메서드에는 지역변수가 없는데 여기에만 지역변수를 생성한 이유가 따로 있는건가요?

ddu0422 commented on src/main/java/woowahan/anifarm/tecolearning/study/domain/Study.java:
```diff
@@ -12,7 +13,7 @@
 @Getter
 @ToString
-public class Study {
+public class Study extends BaseEntity {
```
👍

ddu0422 reviewed (commented):
스터디 도메인 구현하시느라 고생많으셨어요 ㅎㅎ
피드백 남겼습니다 !

추가로 study.service 패키지에 dto와 service가 섞여있는데 dto 패키지를 만들어서 분리하는건 어떤가요?

Deocksoo committed:
refactor: StudyService.findPageOfInfoDto에서 불변 collection 반환하도록 수정

Deocksoo committed:
refactor: SutdyService에 StudySummaryDto추가

Study 정보를 page단위로 조회할 때, 요약된 정보만 추려서 제한하여 보여줄 수 있도록
새로운 Dto를 생생하였습니다.

Deocksoo committed:
refactor: SutdyApiController 개행 오타 수정

Deocksoo committed:
feat: LoggedInConfig에서 /api/study-summary를 exclude 하도록 설정

StudyApiController에서 해당 요청을 처리해주는 경우 LoggedInConfig에서 exclude설정하는 데에
어려움이 있었습니다. 따라서 로그인이 필요하지 않은 요청의 경우 StudySummaryController에서 별도의
api를 통해 제공해주도록 수정하였습니다.

Deocksoo committed:
refactor: StudyInfoDto totalNumberOfParticipants 변수명 -> totalNumberOfRecruitment

Deocksoo committed:
refactor: /api/study-summary path를 /api/studies/summary로 변경
